From 8ce7a37070e1f6bdba2bba580a69d85298a62768 Mon Sep 17 00:00:00 2001
From: suve <veg@svgames.pl>
Date: Fri, 20 Jan 2023 15:00:10 +0100
Subject: [PATCH] Add missing cstdint includes

Some files have been using various fixed-size types (uint32_t etc.),
which are defined in stdint.h / cstdint, without including said header
file. While this code worked with GCC12 (likely a transitive include),
under GCC13 it fails to build due to "unknown type" errors.
---
 src/Format.cpp              | 1 +
 src/common/Platform.cpp     | 1 +
 src/graphics/FontReader.cpp | 1 +
 src/gui/font/FontEditor.cpp | 1 +
 4 files changed, 4 insertions(+)

diff --git a/src/Format.cpp b/src/Format.cpp
index 6bbcc4b306..e320cf9c50 100644
--- a/src/Format.cpp
+++ b/src/Format.cpp
@@ -7,6 +7,7 @@
 #include <cstring>
 #include <zlib.h>
 #include <cstdio>
+#include <cstdint>
 
 #include "graphics/Graphics.h"
 
diff --git a/src/common/Platform.cpp b/src/common/Platform.cpp
index b0230f0e60..05ed4af4ae 100644
--- a/src/common/Platform.cpp
+++ b/src/common/Platform.cpp
@@ -25,6 +25,7 @@
 # include <dirent.h>
 #endif
 #ifdef MACOSX
+# include <cstdint>
 # include <mach-o/dyld.h>
 #endif
 
diff --git a/src/graphics/FontReader.cpp b/src/graphics/FontReader.cpp
index ff09eb7519..4141a11cc6 100644
--- a/src/graphics/FontReader.cpp
+++ b/src/graphics/FontReader.cpp
@@ -4,6 +4,7 @@
 #include "font.bz2.h"
 
 #include <array>
+#include <cstdint>
 
 unsigned char *font_data = nullptr;
 unsigned int *font_ptrs = nullptr;
diff --git a/src/gui/font/FontEditor.cpp b/src/gui/font/FontEditor.cpp
index 3db802402e..ea462edb0c 100644
--- a/src/gui/font/FontEditor.cpp
+++ b/src/gui/font/FontEditor.cpp
@@ -1,4 +1,5 @@
 #include <stdexcept>
+#include <cstdint>
 #include <fstream>
 #include <iterator>
 #include <iomanip>
